app/scripts/lib/multichain-api/wallet-requestPermissions.test.js

```diff
@@ -0,0 +1,369 @@
+          {
+            type: Caip25CaveatType,
+            value: {
+              requiredScopes: {
+                'eip155:1': {
+                  methods: [],
+                  notifications: [],
+                  accounts: ['eip155:1:0x1', 'eip155:1:0x2'],
+                },
+                'eip155:5': {
+                  methods: [],
+                  notifications: [],
+                  accounts: ['eip155:5:0x1', 'eip155:5:0x3'],
+                },
+              },
+              optionalScopes: {
+                'eip155:1': {
+                  methods: [],
+                  notifications: [],
+                  accounts: ['eip155:1:0x4'],
+                },
+                'other:1': {
+                  methods: [],
+                  notifications: [],
+                  accounts: ['other:1:0x4'],
+                },
+              },
+            },
+          },
+        ],
+      },
+      otherPermission: {
+        id: '3',
+        parentCapability: 'otherPermission',
+        caveats: [
+          {
+            value: {
+              foo: 'bar',
+            },
+          },
+        ],
+      },
+    }),
+  );
+  const getNetworkConfigurationByNetworkClientId = jest.fn().mockReturnValue({
+    chainId: '0x1',
+  });
+  const updateCaveat = jest.fn();
+  const grantPermissions = jest.fn().mockReturnValue(
+    Object.freeze({
+      [Caip25EndowmentPermissionName]: {
+        id: 'new',
+        parentCapability: Caip25EndowmentPermissionName,
+        caveats: [
+          {
+            type: Caip25CaveatType,
+            value: {
+              requiredScopes: {},
+              optionalScopes: {},
+            },
+          },
+        ],
+      },
+    }),
+  );
+  const response = {};
+  const handler = (request) =>
+    requestPermissionsHandler.implementation(request, response, next, end, {
+      requestPermissionsForOrigin,
+      getPermissionsForOrigin,
+      getNetworkConfigurationByNetworkClientId,
+      updateCaveat,
+      grantPermissions,
+    });
+
+  return {
+    response,
+    next,
+    end,
+    requestPermissionsForOrigin,
+    getPermissionsForOrigin,
+    getNetworkConfigurationByNetworkClientId,
+    updateCaveat,
+    grantPermissions,
+    handler,
+  };
+};
+
+describe('requestPermissionsHandler', () => {
+  it('returns an error if params is malformed', async () => {
+    const { handler, end } = createMockedHandler();
+
+    const malformedRequest = {
+      ...baseRequest,
+      params: [],
+    };
+    await handler(malformedRequest);
+    expect(end).toHaveBeenCalledWith(
+      invalidParams({ data: { request: malformedRequest } }),
+    );
+  });
+
+  it('requests permissions from params, but ignores CAIP-25 if specified', async () => {
+    const { handler, requestPermissionsForOrigin } = createMockedHandler();
+
+    await handler(baseRequest);
+    expect(requestPermissionsForOrigin).toHaveBeenCalledWith({
+      eth_accounts: {},
+      otherPermission: {},
```

Comment:
Any reason we shouldn't filter out other permission requests we don't explicitly recognize?

Reply:
ah maybe it doesn't really make sense or matter

Reply:
That seems like a separate concern. My goal with these changes is to translate `caip25:endowment` to `eth_accounts` on read and vice versa on writes